Without pkg-config installed, build fails

I started with a fresh debian 9 install, the first thing I did after installing sudo was start installing kak. I ran into this issue, and it was not very clear what the issue was. I did figure it out after reading through the Makefile.

### Steps <!-- how to reproduce the issue? -->
1. Do not have pkg-config installed.
2. Follow the commands to install kak using Ubuntu (the same commands work for debian).

### Outcome <!-- what is the result of the above steps? -->
During make, the build fails on ncurses_ui.cc: `fatal error: ncurses.h: No such file or directory`
This is because pkg-config does not return a valid list of libraries to be included for compilation.

### Expected <!-- what should have been the result of the above steps? -->
Some indication that pkg-config is the issue, and not ncurses.h or ncurses_ui.cc

According to the error you're showing, the issue seems to be that you didn't install `libncursesw-dev`. If a program wasn't found in the PATH at the time of the execution of the command by `make`, you will already get an error saying just that:

```
make: pkg-config: Command not found
make: *** [Makefile:2: all] Error 127
```

I see what happened now: `make` does show a non-critical error if `pkg-config` isn't in the PATH, it's the very first line displayed but is somewhat overshadowed by the subsequent compilation commands. However, since the call is only made within the scope of a variable initialization, the `all` target will be executed anyway:

```
lenormf@5d157d4e0f79:~/kakoune/src$ make
make: pkg-config: Command not found

[…]

ncurses_ui.cc:14:21: fatal error: ncurses.h: No such file or directory
 #include <ncurses.h>
                     ^
compilation terminated.
Makefile:80: recipe for target '.ncurses_ui.debug.o' failed
make: *** [.ncurses_ui.debug.o] Error 1
```

I've made a pull request to address that.

--- a/src/Makefile
+++ b/src/Makefile
@@ -24,7 +24,7 @@ sources := $(sort $(wildcard *.cc))
 objects := $(addprefix ., $(sources:.cc=$(suffix).o))
 deps := $(addprefix ., $(sources:.cc=$(suffix).d))
 
-PKG_CONFIG ?= pkg-config
+PKG_CONFIG ?= $(shell command -v pkg-config 2>/dev/null)
 
 PREFIX ?= /usr/local
 DESTDIR ?= # root dir
@@ -66,7 +66,12 @@ endif
 
 CXXFLAGS += -pedantic -std=gnu++14 -g -Wall -Wextra -Wno-unused-parameter -Wno-reorder -Wno-sign-compare -Wno-address -Wno-noexcept-type -Wno-unknown-attributes -Wno-unknown-warning-option
 
-all : kak
+all : check_pkg_config kak
+
+check_pkg_config:
+ifeq ($(PKG_CONFIG),)
+	$(error "pkg-config not found in PATH")
+endif
 
 kak : kak$(suffix)
 	ln -sf $< $@

